include/albatross/src/models/ransac_gp.hpp

```diff
@@ -15,176 +15,128 @@
-                    const Eigen::MatrixXd &) const {
+  double operator()(const JointDistribution &prior,
+                    const MarginalDistribution &truth) const {
     // Negative because a lower metric is better.
-    return (-1.0 * static_cast<double>(indices.size()));
+    return (-1.0 * static_cast<double>(truth.size()));
   }
 };
 
 struct ChiSquaredConsensusMetric {
-  double operator()(const GroupIndices &indices,
-                    const MarginalDistribution &targets,
-                    const Eigen::MatrixXd &cov) const {
-    const auto consensus_prior = symmetric_subset(cov, indices);
-    const auto consensus_targets = subset(targets, indices);
-    return chi_squared_cdf(consensus_targets.mean, consensus_prior);
+  double operator()(const JointDistribution &prior,
+                    const MarginalDistribution &truth) const {
+    return chi_squared_cdf(prior, truth);
   }
 };
 
 struct ChiSquaredIsValidCandidateMetric {
-
-  template <typename FeatureType>
-  bool operator()(const GroupIndices &inds,
-                  const RegressionDataset<FeatureType> &dataset,
-                  const Eigen::MatrixXd &cov) const {
-    const auto train_dataset = subset(dataset, inds);
-    const auto train_cov = symmetric_subset(cov, inds);
-
-    const JointDistribution prior(Eigen::VectorXd::Zero(train_cov.rows()),
-                                  train_cov);
+  bool operator()(const JointDistribution &pred,
+                  const MarginalDistribution &truth) const {
     // These thresholds are under the assumption of a perfectly
     // representative prior.
-    const double probability_prior_exceeded =
-        chi_squared_cdf(prior, train_dataset.targets);
+    const double probability_prior_exceeded = chi_squared_cdf(pred, truth);
     const double skip_every_1000th_candidate = 0.999;
     return (probability_prior_exceeded < skip_every_1000th_candidate);
   };
 };
 
 struct AlwaysAcceptCandidateMetric {
-  template <typename FeatureType>
-  bool operator()(const GroupIndices &inds,
-                  const RegressionDataset<FeatureType> &dataset,
-                  const Eigen::MatrixXd &cov) const {
+  bool operator()(const JointDistribution &pred,
+                  const MarginalDistribution &truth) const {
     return true;
   }
 };
 
-template <typename ModelType, typename FeatureType, typename InlierMetric,
-          typename ConsensusMetric, typename IsValidCandidateMetric,
-          typename GroupKey>
-inline RansacFunctions<FitAndIndices<ModelType, FeatureType>, GroupKey>
-get_gp_ransac_functions(
-    const ModelType &model,
-    const RegressionDataset<FeatureType> &dataset_with_mean,
+template <typename InlierMetric, typename ConsensusMetric,
+          typename IsValidCandidateMetric, typename GroupKey>
+inline RansacFunctions<ConditionalFit, GroupKey> get_gp_ransac_functions(
+    const JointDistribution &prior, const MarginalDistribution &truth,
```

Comment:
Any pros/cons for having  `const ConditionalGaussian &model` instead of `const JointDistribution &prior, const MarginalDistribution &truth,` in this functions arguments?

Reply:
I don't see any critical pros/cons.  My thought was that it's probably best to keep the `ConditionalGaussian` hidden from the user.  In otherwords, I think that for someone using `albatross` the should be familiar with `JointDistribution` and `MarginalDistribution` and should understand how to produce these things, but they may not know about the `ConditionalGaussian` model.